save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/LnkUserOrganizationController.kt

```diff
@@ -193,6 +194,27 @@ class LnkUserOrganizationController(
         )
         return (exactMatchUsers + prefixUsers).map { it.toUserInfo() }
     }
+
+    /**
+     * Get not deleted organizations that are connected with current user.
+     *
+     * @param authentication
+     * @return Map where key is organization name, value is a pair of avatar and role.
+     */
+    @Suppress("TYPE_ALIAS", "UnsafeCallOnNullableType")
+    @GetMapping("/by-user/not-deleted")
+    fun getOrganizationWithRoles(
+        authentication: Authentication,
+    ): Map<String, Pair<String?, Role>> {
+        val selfId = (authentication.details as AuthenticationDetails).id
+        val user = lnkUserOrganizationService.getUserById(selfId).get()
+        return lnkUserOrganizationService
+            .getOrganizationsAndRolesByUser(user)
+            .filter { it.organization?.status != OrganizationStatus.DELETED }
+            .associate {
+                (it.organization!!.name) to (it.organization!!.avatar to (it.role ?: Role.NONE))
```

Comment:
Let's create a data class for this. Even if it's just a couple of fields, it will be easier in the future if it is declared as a statically typed object

Reply:
Done